app/Http/Controllers/Dashboard/DashboardController.php

```diff
@@ -79,9 +79,26 @@ public function showDashboard()
         $components = Component::orderBy('order')->get();
         $incidents = $this->getIncidents();
         $subscribers = $this->getSubscribers();
-        $usedComponentGroups = Component::enabled()->where('group_id', '>', 0)->groupBy('group_id')->pluck('group_id');
-        $componentGroups = ComponentGroup::whereIn('id', $usedComponentGroups)->orderBy('order')->get();
-        $ungroupedComponents = Component::enabled()->where('group_id', 0)->orderBy('order')->orderBy('created_at')->get();
+
+        // Component & Component Group lists.
+        $usedComponentGroups = Component::enabled()->where('group_id', '>', 0)
+            ->groupBy('group_id')
+            ->pluck('group_id')
+        ;
+        $componentGroupsBuilder = ComponentGroup::guest();
+        if (auth()->check()) {
+            $componentGroupsBuilder = ComponentGroup::loggedIn(auth()->user());
+        }
+        $componentGroups = $componentGroupsBuilder->whereIn('id', $usedComponentGroups)
+            ->orderBy('order')
+            ->get()
+        ;
+        $ungroupedComponents = Component::enabled()
+            ->where('group_id', 0)
+            ->orderBy('order')
+            ->orderBy('created_at')
+            ->get()
+        ;
```

Comment:
They're a total waste of time and probably arn't configured to the exact specification we want. I'd discourage them everywhere anyway because that statement is generally true. I'd recommend using StyleCI to enforce CS. Just turn it on and forget. It'll merge fixes for you.